src/ray/core_worker/test/mock_worker.cc

```diff
@@ -21,111 +21,111 @@ using namespace std::placeholders;
 
 namespace ray {
 
-/// A mock C++ worker used by core_worker_test.cc to verify the task submission/execution
-/// interfaces in both single node and cross-nodes scenarios. As the raylet client can
-/// only
-/// be called by a real worker process, core_worker_test.cc has to use this program binary
-/// to start the actual worker process, in the test, the task submission interfaces are
-/// called
-/// in core_worker_test, and task execution interfaces are called in this file, see that
-/// test
-/// for more details on how this class is used.
-class MockWorker {
- public:
-  MockWorker(const std::string &store_socket, const std::string &raylet_socket,
-             int node_manager_port, const gcs::GcsClientOptions &gcs_options)
-      : worker_(WorkerType::WORKER, Language::PYTHON, store_socket, raylet_socket,
-                JobID::FromInt(1), gcs_options, /*log_dir=*/"",
-                /*node_id_address=*/"127.0.0.1", node_manager_port,
-                std::bind(&MockWorker::ExecuteTask, this, _1, _2, _3, _4, _5, _6, _7)) {}
-
-  void StartExecutingTasks() { worker_.StartExecutingTasks(); }
+int64_t prev_seq_no_ = 0;
 
- private:
-  Status ExecuteTask(TaskType task_type, const RayFunction &ray_function,
-                     const std::unordered_map<std::string, double> &required_resources,
-                     const std::vector<std::shared_ptr<RayObject>> &args,
-                     const std::vector<ObjectID> &arg_reference_ids,
-                     const std::vector<ObjectID> &return_ids,
-                     std::vector<std::shared_ptr<RayObject>> *results) {
-    // Note that this doesn't include dummy object id.
-    const ray::FunctionDescriptor function_descriptor =
-        ray_function.GetFunctionDescriptor();
-    RAY_CHECK(function_descriptor->Type() ==
-              ray::FunctionDescriptorType::kPythonFunctionDescriptor);
-    auto typed_descriptor = function_descriptor->As<ray::PythonFunctionDescriptor>();
+Status GetWorkerPid(std::vector<std::shared_ptr<RayObject>> *results) {
```

Comment:
why move this function out of `MockWorker`? Is it related to this PR?

Reply:
I refactored this file to use C functions instead of a C++ class. However, it makes the changes hard to review. I've reverted the refactoring.